activejob/test/cases/attributes_test.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "helper"
+
+class AccountScopedJob < ActiveJob::Base
+  attribute :account_id
+
+  def perform
+    JobBuffer.add(account_id)
+  end
+end
+
+class SubClassedJob < AccountScopedJob
+end
+
+class AttributesTest < ActiveSupport::TestCase
+  setup do
+    JobBuffer.clear
+  end
+
+  test "serializes and deserializes the defined attributes" do
```

Comment:
Could we have a test serializing a complex attribute? (array, hash, Active Record / GlobalID-able)?